compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -883,22 +883,6 @@ impl<'tcx> ty::FallibleTypeFolder<TyCtxt<'tcx>> for RemapHiddenTyRegions<'tcx> {
         self.tcx
     }
 
-    fn try_fold_ty(&mut self, t: Ty<'tcx>) -> Result<Ty<'tcx>, Self::Error> {
```

Comment:
We don't need this. 

In fact, it has the potential to cause ICEs since us *not* remapping bivariant lifetimes on opaques may lead us to leaking infer vars into query responses. This does not happen in practice though.